test/fixture/person.scrypt

```diff
@@ -0,0 +1,37 @@
+struct Person {
+    bytes addr;
+    bool isMale;
+    int age;
+}
+
+struct Block {
+    bytes hash;
+    bytes header;
+    int time;
+}
+
+contract Main {
+    struct Person person;
+    int x;
+
+    constructor(struct Person some, int x) {
+        this.person = some;
+        this.x = x;
+    }
+
+    public function main(struct Person p, int x, bool isMale) {
+
+        require(p.isMale == isMale);
+        require(p.age > x);
+        require(! p.isMale);
+
+        struct Person c = this.incAge(p);
+        require(c.age == 34);
```

Comment:
struct Person c = this.incAge(p);
require(c.age == 34);
require(p.age == 33);


p = this.incAge(p);
require(p.age == 34);

Reply:
6896aa57a96d35ef0203f2c957914ffb2bb37688